test.cpp

```diff
@@ -29,6 +29,36 @@ TEST(Lexer, HelloWorld) {
   ASSERT_EQ(lexer.Lex().get(), xos::Token(xos::Token::eof, 2, 20));
 }
 
+TEST(Lexer, ExpectEndlineCharacterAtLastChar) {
+  char helloWorldStr[] = "\"Hello World\\n\"";
+  std::stringstream ss(helloWorldStr);
```

Comment:
Note that passing a `char[]` to the ctor here will create a temporary `std::string`. Let's instead just create the stringstream and pipe the hello world string into it:

```
std::stringstream ss;
ss << kHelloWorldStr;
```

We should also apply these for the tests below.